src/compiler/test-file/base.js

```diff
@@ -19,8 +19,8 @@ export default class TestFileCompilerBase {
         throw new Error('Not implemented');
     }
 
-    canCompile (code, filename) {
-        return this.supportedExtensionRe.test(filename) && this._hasTests(code);
+    canCompile (code, filename, disableTestSyntaxValidation) {
+        return this.supportedExtensionRe.test(filename) && (disableTestSyntaxValidation || this._hasTests(code));
```

Comment:
You can just use `this.disableTestSyntaxValidation` here.

Reply:
`this.disableTestSyntaxValidation` is defined in `Compiler` which is not based class for `EsNextTestFileCompiler`, `TypeScriptTestFileCompiler` etc.